pages/post/[postId].js

```diff
@@ -22,18 +22,16 @@ export async function getServerSideProps(context) {
     let post = await GetstreamSingleton.getInstance().getPostById(originalPostId)
 
     if (!PostUtil.isPostPublic(post)) {
-        console.log('masuk private')
         let redirect = await RedirectUtils.redirectPrivatePost(userAgent, post)
         if (redirect) return redirect
     }
 
     if (DateUtils.isPostExpired(post)) {
-        console.log('masuk expired')
         let redirect = await RedirectUtils.redirectExpiredPost(userAgent)
         if (redirect) return redirect
     }
 
-    if (UserAgentUtils.isMobile(userAgent)) {
+    if (UserAgentUtils.isMobile(userAgent)) {        
         let redirect = await RedirectUtils.redirectMobileDevice(userAgent, post)
         if (redirect && !isDynamicLink) return redirect
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The condition `if (redirect && !isDynamicLink)` might lead to unexpected behavior if `isDynamicLink` is not defined or null. It would be better to ensure that `isDynamicLink` is a boolean value before using it in the condition.
```diff
-    if (UserAgentUtils.isMobile(userAgent)) {        
+    if (UserAgentUtils.isMobile(userAgent) && typeof isDynamicLink === 'boolean') {        
         let redirect = await RedirectUtils.redirectMobileDevice(userAgent, post)
         if (redirect && !isDynamicLink) return redirect
     }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->